src/arrayMethodReduce.js

```diff
@@ -5,7 +5,23 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let reducedValue = initialValue;
+    let startIndex = 0;
+
+    if (!(typeof callback === 'function')) {
+      return undefined;
```

Comment:
А ще краще викидати помилку і зупинити виконання коду
```suggestion
      return ;
```